### PR TITLE
Add a success message when all checks are passed and a summary of number of files scanned

### PR DESCRIPTION
**Pull request by jatkinson1000**
When running `fortitude check` with everything passing there is no information printed to stdout.
I thought it would be nice to have some confirmation that everything had been passed and the scanned code is fully compliant.

At the same time, when running on a large project, I thought it might be nice to have a summary of how many files have been scanned, so have also implemented this.

If you think this is a useful addition I would love to have it merged into main.
I realise message formatting and content is a matter of taste, so am happy to adjust the messages as you feel would be most appropriate - please feel free to make suggestions/requests in a review.

In summary, a successful run of `fortitude check` will now write:
```
[user@machine code]$ fortitude check src/myfile.f90

fortitude: 1 files scanned.
All checks passed!

[user@machine code]$
```

I also added the summary of number of files scanned to the failing output:
```
[user@machine code]$ fortitude check src/

fortitude: 64 files scanned.
-- Number of errors: 375
-- For more information, run:

    fortitude explain [ERROR_CODES]

[user@machine code]$ 
```

### CODE REVIEW

**Review by LiamPattinson — COMMENTED** [review]
Overall, I like it! Initially I approached this with the opinion that a successful run should print nothing and return 0, like flake8, but as this is indistinguishable from the program not doing anything at all, I'm coming around to the idea that it should always report something.

When it does report errors, I think the lines beginning with `--` now look a bit awkward (although in hindsight they probably always did...). Maybe it could be tweaked to print something like this?

```
fortitude: 82 files scanned. 12383 errors found.
For more information, run:

    fortitude explain [ERROR_CODES]

```

File: fortitude/src/check.rs [hunk 1]
```diff
@@ -197,14 +198,21 @@ pub fn check(args: CheckArgs) -> Result<ExitCode> {
                         total_errors += 1;
                     }
                 }
+                total_files += 1;
             }
+            let file_no = format!("fortitude: {} files scanned.", total_files);
```
[the remark this answers is not in the document]
Comment (jatkinson1000) [reply fortitude/src/check.rs]: This wasn't cargo fmt compliant, so I have amended the commit https://github.com/PlasmaFAIR/fortitude/pull/104/commits/ad45e498742fc657b841e529635ce671a8190e5d to include the suggestion.

**Comment by jatkinson1000** [comment]
Thanks.

I agree that some positive confirmation is useful.
ruff gives `all checks passed!`, and black does that thing with the cake 🍰.

Commit https://github.com/PlasmaFAIR/fortitude/pull/104/commits/3bd4b684d9d7f343469efcdb08989ac42975455f adjusts the error message to read:
```
[user@machine code]$ fortitude check test.f90

...

fortitude: 1 files scanned.
Number of errors: 33
For more information, run:

    fortitude explain [ERROR_CODES]

[user@machine code]$
```

File: fortitude/src/check.rs [hunk 2]
```diff
@@ -197,14 +198,21 @@ pub fn check(args: CheckArgs) -> Result<ExitCode> {
...
             if total_errors == 0 {
+                let success = "All checks passed!".bright_green();
+                println!("\n{}\n{}\n", file_no, success);
                 Ok(ExitCode::SUCCESS)
             } else {
                 let err_no = format!("Number of errors: {}", total_errors.to_string().bold());
                 let info = "For more information, run:";
                 let explain = format!("{} {}", "fortitude explain", "[ERROR_CODES]".bold());
-                println!("\n-- {}\n-- {}\n\n    {}\n", err_no, info, explain);
+                println!("\n{}\n{}\n\n{}\n\n    {}\n", file_no, err_no, info, explain);
```
Comment (ZedThree) [inline fortitude/src/check.rs:215]: If we're touching this, may as well use the f-string style syntax?

```suggestion
                println!("\n{file_no}\n{err_no}\n\n{info}\n\n    {explain}\n");
```

File: fortitude/src/check.rs [hunk 3]
```diff
@@ -197,14 +198,21 @@ pub fn check(args: CheckArgs) -> Result<ExitCode> {
...
                 }
+                total_files += 1;
             }
+            let file_no = format!(
+                "fortitude: {} files scanned.",
+                total_files.to_string().bold()
+            );
             if total_errors == 0 {
+                let success = "All checks passed!".bright_green();
+                println!("\n{}\n{}\n", file_no, success);
```
Comment (ZedThree) [inline fortitude/src/check.rs:209]:
```suggestion
                println!("\n{file_no}\n{success}\n");
```

**Review by ZedThree — APPROVED** [review]
LGTM, thanks @jatkinson1000 ! A couple of very minor comments.
